### PR TITLE
Use Tag for Token File OutPut

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Update Swagger Parser to save file using the tag used for generation the APIView token

### CODE REVIEW

**Comment by maririos** [comment]
Curious on why we need this change?

File: tools/apiview/parsers/swagger-api-parser/SwaggerApiParserTest/ReadmeParserTest.cs [hunk 1]
```diff
@@ -77,7 +79,8 @@ public void TestGetTagFromYamlArgumentsInvalidCase()
     public void TestOrderedInputFiles()
     {
         const string readmeFilePath = "./fixtures/unordered.md";
-        var inputFiles = ReadmeParser.GetSwaggerFilesFromReadme(readmeFilePath, "package-2023-02");
+        var tag = "package-2023-02";
```
Comment (maririos) [inline tools/apiview/parsers/swagger-api-parser/SwaggerApiParserTest/ReadmeParserTest.cs:82]: could you add tests for the feature you are adding?

**Comment by chidozieononiwu** [comment]
> Curious on why we need this change?

So this tool has a logic for identifying the default tag. Am trying to avoid reimplementing that logic in the script.
This provides a way to get the tag used to create the APIView artifact so the same tag can be used for the Baseline.

**Comment by praveenkuttappan** [comment]
What if the change is not in default tag? It's possible to have swagger readme with default tag with unchanged files but non default has changed files. So I believe script was identifying the correct tag earlier to be used to create APIView token file.

File: tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs [hunk 2]
```diff
@@ -114,6 +118,13 @@ static async Task HandleGenerateCodeFile(IEnumerable<string> swaggers, string ou
 
             var codeFile = root.GenerateCodeFile();
             var outputFilePath = Path.GetFullPath(output);
+
+            if (useTagForOutput)
+            {
+                output = $"{readmeTag}.json";
```
Comment (praveenkuttappan) [inline tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs:124]: Where do you rename this with new or baseline postfix? how is it handled when both source and target has same tag?

**Review by praveenkuttappan — APPROVED** [review]

File: tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs (hunk 2 above)
Comment (chidozieononiwu) [reply tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs:124]: https://github.com/Azure/azure-rest-api-specs/blob/00d50cba07376c494f3b50647b847cd4ba348a04/eng/scripts/Create-APIView.ps1#L118

**Pull request merged**